src/Libraries/Microsoft.Extensions.DataIngestion.MarkItDown/MarkItDownReader.cs

```diff
@@ -0,0 +1,121 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Shared.Diagnostics;
+
+namespace Microsoft.Extensions.DataIngestion;
+
+/// <summary>
+/// Reads documents by converting them to Markdown using the <see href="https://github.com/microsoft/markitdown">MarkItDown</see> tool.
+/// </summary>
+public class MarkItDownReader : IngestionDocumentReader
+{
+    private readonly string _exePath;
+    private readonly bool _extractImages;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="MarkItDownReader"/> class.
+    /// </summary>
+    /// <param name="exePath">The path to the MarkItDown executable. When not provided, "markitdown" needs to be added to PATH.</param>
+    /// <param name="extractImages">A value indicating whether to extract images.</param>
+    public MarkItDownReader(string exePath = "markitdown", bool extractImages = false)
+    {
+        _exePath = Throw.IfNullOrEmpty(exePath);
+        _extractImages = extractImages;
+    }
+
+    /// <inheritdoc/>
+    public override async Task<IngestionDocument> ReadAsync(FileInfo source, string identifier, string? mediaType = null, CancellationToken cancellationToken = default)
+    {
+        _ = Throw.IfNull(source);
+        _ = Throw.IfNullOrEmpty(identifier);
+
+        if (!source.Exists)
+        {
+            throw new FileNotFoundException("The specified file does not exist.", source.FullName);
+        }
+
+        ProcessStartInfo startInfo = new()
+        {
+            FileName = _exePath,
```

Comment:
Is running an EXE by name from the working directory considered safe?  I would have expected us to run from application base, if that's where we install the tool, or at least where we consider to be a "trusted location" for runnable code.

Reply:
This looks reasonable.  Thank you for addressing this.